executor/src/witgen/machines/machine_extractor.rs

```diff
@@ -32,6 +30,36 @@ pub struct ExtractionOutput<'a, T: FieldElement> {
     pub base_parts: MachineParts<'a, T>,
 }
 
+#[derive(Default)]
+/// Keeps track of the global set of publics that are referenced by the machine's identities.
+struct PublicsTracker<'a>(BTreeSet<&'a String>);
+
+impl<'a> PublicsTracker<'a> {
+    /// Given a machine's identities, add all publics that are referenced by them.
+    /// Panics if a public is referenced by more than one machine.
+    fn add_all<T>(
+        &mut self,
+        identities: &[&'a powdr_ast::analyzed::Identity<SelectedExpressions<Expression<T>>>],
+    ) {
+        let referenced_publics = identities
```

Comment:
I don't see a proper use-case, but actually we should only consider LHSs of lookups.

Reply:
Ah, but all the identities here should be non-lookups, ok.